Tasks/DockerV2/docker.ts

```diff
@@ -2,16 +2,27 @@
 
 import path = require('path');
 import * as tl from "vsts-task-lib/task";
+import AuthenticationTokenProvider  from "docker-common/registryauthenticationprovider/authenticationtokenprovider";
+import ACRAuthenticationTokenProvider from "docker-common/registryauthenticationprovider/acrauthenticationtokenprovider"
 import GenericAuthenticationTokenProvider from "docker-common/registryauthenticationprovider/genericauthenticationtokenprovider";
 import RegistryAuthenticationToken from "docker-common/registryauthenticationprovider/registryauthenticationtoken";
 import ContainerConnection from 'docker-common/containerconnection';
-import Q = require('q');
 
 tl.setResourcePath(path.join(__dirname, 'task.json'));
 
-let containerRegisitry = tl.getInput("containerRegistry");
-let authenticationProvider = new GenericAuthenticationTokenProvider(containerRegisitry);        
-let registryAuthenticationToken = authenticationProvider.getAuthenticationToken();
+let endpointId = tl.getInput("containerRegistry");
+let authenticationProvider: AuthenticationTokenProvider;
+
+const registryType: string = tl.getEndpointDataParameter(endpointId, "registrytype", true);
+if(registryType ==  "ACR"){
+    const loginServer = tl.getEndpointAuthorizationParameter(endpointId, "loginServer", false);
+    authenticationProvider = new ACRAuthenticationTokenProvider(endpointId, loginServer);
+}
+else {
+    authenticationProvider = new GenericAuthenticationTokenProvider(endpointId);
+}
```

Comment:
Have only 1 copy of this code.

Reply:
Ok sure. Will refactor in the immediate next PR